src/main/java/com/spartronics4915/frc2019/subsystems/PanelHandler.java

```diff
@@ -88,21 +92,33 @@ public void onLoop(double timestamp)
                 SystemState newState = defaultStateTransfer();
                 switch (mSystemState)
                 {
-                    case RETRACTING:
+                    case ARM_DOWNING:
                         if (mStateChanged)
                         {
-                            mSolenoid.set(Constants.kPanelSolenoidRetract);
+                            mSolenoidArm.set(Constants.kPanelSolenoidRetract);
+                        }
+                    case ARM_UPING:
+                        if (mStateChanged)
+                        {
+                            mSolenoidArm.set(Constants.kPanelSolenoidExtend);
                         }
-                        break;
-                    case EJECTING://BB6
+                    case EJECTING_ROCKET:
                         if (mStateChanged)
                         {
+                            mSolenoidArm.set(Constants.kPanelSolenoidRetract);
+                            mArmDownTime = Timer.getFPGATimestamp();
+                        }
+                        else if (Timer.getFPGATimestamp() > mArmDownTime + Constants.kPanelArmDownTime && newState == mSystemState){
```

Comment:
No need for `newState == mSystemState` here because you're not changing `mSystemState`.